demo/src/main/java/com/smashingboxes/surelockdemo/LoginActivity.kt

```diff
@@ -0,0 +1,346 @@
+import android.content.Context
+import android.content.SharedPreferences
+import android.os.AsyncTask
+import android.os.Bundle
+import android.support.v7.app.AlertDialog
+import android.support.v7.app.AppCompatActivity
+import android.text.TextUtils
+import android.util.Log
+import android.view.View
+import android.view.inputmethod.EditorInfo
+import android.widget.Button
+import android.widget.TextView
+import android.widget.Toast
+import com.smashingboxes.surelock.*
+import kotlinx.android.synthetic.main.activity_login.*
+import java.io.UnsupportedEncodingException
+
+/**
+ * Created by Tyler McCraw on 2/17/17.
+ *
+ *
+ * A login screen that offers login via username/password
+ * along with the option to set up future fingerprint authentication.
+ *
+ */
+
+class LoginActivity : AppCompatActivity(), SurelockFingerprintListener {
+
+    private var authTask: UserLoginTask? = null
+    private var surelock: Surelock? = null
+    private lateinit var surelockStorage: SurelockStorage
+    private var preferences: SharedPreferences? = null
+
+    private val preferenceHelper: SharedPreferences
+        get() {
+            if (preferences == null) {
+                preferences = getSharedPreferences(SHARED_PREFS_FILE_NAME, Context.MODE_PRIVATE)
+            }
+            return preferences!!
+        }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_login)
+        surelockStorage = SharedPreferencesStorage(this, SHARED_PREFS_FILE_NAME)
+        setUpViews()
+    }
+
+    private fun setUpViews() {
+        password.setOnEditorActionListener(
+            TextView.OnEditorActionListener { _, id, _ ->
+                if (id == R.id.login || id == EditorInfo.IME_NULL) {
+                    handleLoginClick()
+                    return@OnEditorActionListener true
+                }
+                false
+            })
+
+        val signInButton = findViewById<View>(R.id.sign_in_button) as Button
+        signInButton.setOnClickListener { handleLoginClick() }
+    }
+
+    override fun onResume() {
+        super.onResume()
+        if (userWantsToUseFingerprintToLogin()) {
+            if (!Surelock.fingerprintAuthIsSetUp(this, true)) {
+                fingerprint_checkbox.visibility = View.VISIBLE
+                fingerprint_checkbox.isChecked = false
+            } else {
+                try {
+                    getSurelock().loginWithFingerprint(KEY_CRE_DEN_TIALS)
+                } catch (e: SurelockInvalidKeyException) {
+                    surelockStorage.clearAll()
+                    showFingerprintLoginInvalidated()
+                }
+
+                fingerprint_checkbox.visibility = View.GONE
+            }
+        }
+    }
+
+    override fun onStop() {
+        //Clear storage so users don't have to clear app data to try out new credentials
+        surelockStorage.clearAll()
```

Comment:
I would also maybe mention that this is just for demo purposes.
Dummy-proof it for the copy-pasters out there lol